src/room.js

```diff
@@ -0,0 +1,37 @@
+class Room {
+
+  constructor (roomname) {
+    this.roomname = roomname;
+    this.endpoints = {};
+  }
+  addEndpoint (endpointId) {
+    this.endpoints[endpointId] = {
+      name: '',
+      permissions: 'CHAT',
+      commsid: ''
```

Comment:
Rather than initialising endpoints with some empty strings, we could either have a few more methods:
+ addUsername
+ addPermissions
+ addCommsId

Or change the "update" methods so that they can also be used for adding `username` etc when they don't yet exist